test_avoids_zig_zag_trap is flaky

```
=================================== FAILURES ===================================
___________________________ test_avoids_zig_zag_trap ___________________________
[gw5] linux2 -- Python 2.7.14 /usr/bin/python2
Traceback (most recent call last):
  File "/build/python-hypothesis/src/hypothesis-python-3.44.23-py2/tests/quality/test_zig_zagging.py", line 38, in test_avoids_zig_zag_trap
    @example((b'\x02', b'', 258))
  File "/build/python-hypothesis/src/hypothesis-python-3.44.23-py2/src/hypothesis/core.py", line 988, in wrapped_test
    test_runner, test, wrapped_test, settings, arguments, kwargs
  File "/build/python-hypothesis/src/hypothesis-python-3.44.23-py2/src/hypothesis/core.py", line 280, in execute_explicit_examples
    lambda data: test(*arguments, **example_kwargs)
  File "/build/python-hypothesis/src/hypothesis-python-3.44.23-py2/src/hypothesis/executors.py", line 58, in default_new_style_executor
    return function(data)
  File "/build/python-hypothesis/src/hypothesis-python-3.44.23-py2/src/hypothesis/core.py", line 280, in <lambda>
    lambda data: test(*arguments, **example_kwargs)
  File "/build/python-hypothesis/src/hypothesis-python-3.44.23-py2/tests/quality/test_zig_zagging.py", line 78, in test_avoids_zig_zag_trap
    b + hbytes([0]) + b + hbytes([1]) + marker))
  File "/build/python-hypothesis/src/hypothesis-python-3.44.23-py2/src/hypothesis/internal/compat.py", line 398, in __radd__
    assert isinstance(value, compatbytes)
AssertionError
---------------------------------- Hypothesis ----------------------------------
Falsifying example: test_avoids_zig_zag_trap(p=('\x01\x00', '', 20048))
```

--- a/Makefile
+++ b/Makefile
@@ -145,8 +145,9 @@ check-pytest30: $(TOX)
 check-pytest28: $(TOX)
 	$(TOX) --recreate -e pytest28
 
-check-quality: $(TOX)
+check-quality: $(TOX) $(PY27)
 	$(TOX) --recreate -e quality
+	$(TOX) --recreate -e quality2
 
 check-ancient-pip: $(PY273)
 	scripts/check-ancient-pip.sh $(PY273)

--- a/scripts/install.sh
+++ b/scripts/install.sh
@@ -86,7 +86,7 @@ install () {
 for var in "$@"; do
   case "${var}" in
     2.7)
-      install 2.7.11 python2.7
+      install 2.7.14 python2.7
       ;;
     2.7.3)
       install 2.7.3 python2.7.3

--- a/tests/quality/test_zig_zagging.py
+++ b/tests/quality/test_zig_zagging.py
@@ -47,6 +47,8 @@ def problem(draw):
 @given(problem())
 def test_avoids_zig_zag_trap(p):
     b, marker, lower_bound = p
+    b = hbytes(b)
+    marker = hbytes(marker)
 
     n_bits = 8 * (len(b) + 1)
 

--- a/tox.ini
+++ b/tox.ini
@@ -34,6 +34,14 @@ deps=
 commands=
     python -m pytest tests/quality/
 
+
+[testenv:quality]
+basepython=python2.7
+deps=
+    -rrequirements/test.txt
+commands=
+    python -m pytest tests/quality/
+
 [testenv:oldpy27]
 basepython=python2.7.3
 deps=

--- a/tox.ini
+++ b/tox.ini
@@ -35,7 +35,7 @@ commands=
     python -m pytest tests/quality/
 
 
-[testenv:quality]
+[testenv:quality2]
 basepython=python2.7
 deps=
     -rrequirements/test.txt

